elements/lisk-chain/src/state_store/account_store.ts

```diff
@@ -132,6 +158,18 @@ export class AccountStore {
 				}
 			}
 		}
+		for (const deletedAddress of this._deletedKeys) {
+			const initialAccount = this._initialAccountValue.get(deletedAddress);
```

Comment:
User may have modified the account before deleting it, logically it seems `_initialAccountValue` contains only initial account fetched from db, any modification to it will not be persisted in it. If that's true, then user may lost the changes in the account state after deleting.

Reply:
We may add some test case to cover the scenario, if user modified account before deleting, then the state diff must contains the modified account in deleted buffer.

Reply:
yes, that's how it should work. if it's deleted, any changes happened before saving should be ignored and keep the original state in the diff.
Otherwise, when block is deleted it will become invalid state

Reply:
And the case is covered https://github.com/LiskHQ/lisk-sdk/pull/5669/files#diff-5fe7163bd6761e4e768a4332ad906cc5R135